### ISSUE TITLE
Support for Laravel 5

### ISSUE THREAD

**Issue by Butochnikov** [jenssegers/laravel-ab#6]
Please add support for laravel 5

**Comment by jenssegers** [comment]
Laravel 5 is already supported.

**Comment by Butochnikov** [comment]
i'm sorry, but no.

After install, error:

```
>php artisan
[BadMethodCallException]
  Call to undefined method [package]
```

This method is no longer supported:

```
$this->package('jenssegers/ab', 'ab', realpath(__DIR__));
```

https://github.com/jenssegers/laravel-ab/blob/master/src/TesterServiceProvider.php#L25

**Comment by Butochnikov** [comment]
What is your solution?

**Comment by garygreen** [comment]
@jenssegers any idea on support for Laravel 5? Was looking for a A/B split testing package this seems perfect...but no Laravel 5 support :-(

**Comment by jenssegers** [comment]
I will look into it when I can find some spare time :)

**Comment by garygreen** [comment]
I can do a PR if your interested? Should be pretty straight forward adding support really...

In an ideal world, I think this a/b split testing is a perfect case for middleware usage `SplitTestingMiddleware` which will decide what experiment to use etc. But let's get L5 up and running first ;-D

**Comment by jenssegers** [comment]
Yeah indeed. The middleware might be useful for Laravel 5, but will break Laravel 4 compatibility though. Or we could just create a new major version and make it Laravel 5 only.

The main thing that was keeping me from adding L5 compatibility was the configuration loading. I did not really look into that yet.

**Comment by garygreen** [comment]
It's really easy with the config loading. Your just make a call to [`$this->mergeConfigFrom(...)`](http://laravel.com/docs/master/packages#configuration) coupled with a `$this->publishes()` for the config file.

The middleware thing could just be a usage 'option' and not a breaking change.

**Comment by Jamongkad** [comment]
Any news on this? The package is still not yet working for Laravel 5.

**Comment by jenssegers** [comment]
I will take a look this weekend. If anyone wants to do a pull request that would be cool :)

**Comment by Jamongkad** [comment]
That would be great :+1:

**Comment by rafelsanso** [comment]
Hi!

I created a new pull request with the changes to migrate Laravel from 4 to 5. Only need to upgrade Travis to pass the tests.
https://github.com/rafelsanso/laravel-ab

If you think the changes are correct, just missing pull accept the request.

Greetings!

**Comment by chicag0** [comment]
Is this working for L5 now?

**Comment by kenny08gt** [comment]
is this working for Laravel 5.3?

**Comment by zablockijj** [comment]
This is not working for Laravel 5.4.
[Symfony\Component\Debug\Exception\FatalThrowableError]              
  Call to undefined method Illuminate\Foundation\Application::share() 
upon running the publish the config or after including the service provider

**Comment by zablockijj** [comment]
See https://laracasts.com/discuss/channels/laravel/undefined-method-illuminatefoundationapplicationshare-when-upgrading-to-laravel-54

**Comment by bfblackjack** [comment]
This doesn't work with Laravel 5.4 even after changing ->share to ->singleton in the vendor package.

**Comment by bart** [comment]
Any updates?

**Comment by ben182** [comment]
I published a new [package ](https://github.com/ben182/laravel-ab) for ab testing that supports laravel 5.
